src/Element/EntityAutocompleteMultiple.php

```diff
@@ -0,0 +1,278 @@
+ * the Autocomplete widget.
+ *
+ * @see \Drupal\Core\Field\Plugin\Field\FieldWidget\EntityReferenceAutocompleteWidget
+ *
+ * Properties:
+ * - #add_more_title: (optional) Title of the "add more" button.
+ *
+ * Usage example:
+ * @code
+ * $form['my_element'] = [
+ *  '#type' => 'oe_corporate_site_info_entity_autocomplete_multiple',
+ *  '#target_type' => 'node',
+ *  '#default_value' => [
+ *    $entity,
+ *  ],
+ *  '#selection_handler' => 'default',
+ *  '#selection_settings' => [
+ *    'target_bundles' => ['article', 'page'],
+ *   ],
+ * ];
+ * @endcode
+ *
+ * @see \Drupal\Core\Entity\Plugin\EntityReferenceSelection\DefaultSelection
+ * @see \Drupal\Core\Entity\Element\EntityAutocomplete
+ *
+ * @FormElement("oe_corporate_site_info_entity_autocomplete_multiple")
+ */
+class EntityAutocompleteMultiple extends FormElement {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getInfo() {
+    $class = get_class($this);
+    return [
+      '#theme' => 'field_multiple_value_form',
+      '#cardinality_multiple' => TRUE,
+      '#description' => NULL,
+      '#add_more_title' => $this->t('Add another item'),
+      '#element_validate' => [[$class, 'validateEntityAutocompleteItems']],
+      '#process' => [
+        [$class, 'processEntityAutocompleteMultiple'],
+        [$class, 'processAjaxForm'],
+      ],
+    ];
+  }
+
+  /**
+   * Apply multiple value features for the entity autocomplete form element.
+   *
+   * @param array $element
+   *   The element.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   * @param array $complete_form
+   *   The original complete form array.
+   *
+   * @return array
+   *   The array of element.
+   */
+  public static function processEntityAutocompleteMultiple(array &$element, FormStateInterface $form_state, array &$complete_form): array {
+    unset($element['#type']);
+    $element['#tree'] = TRUE;
+    $element['#field_name'] = $field_name = end($element['#array_parents']);
+    $element['#default_value'] = $element['#default_value'] ?? [];
+    $parents = $element['#parents'];
+
+    $field_state = [];
+    if (static::getWidgetState($parents, $field_name, $form_state) === NULL) {
+      $max = count($element['#default_value']);
+      $field_state['items_count'] = $max;
+      static::setWidgetState($parents, $field_name, $form_state, $field_state);
+    }
+    else {
+      $max = static::getWidgetState($parents, $field_name, $form_state)['items_count'];
+    }
+
+    for ($i = 0; $i <= $max; $i++) {
+      $element[$i]['target'] = [
+        '#type' => 'entity_autocomplete',
+        '#title' => t('@title (value @number)', ['@title' => $element['#title'], '@number' => $i + 1]),
+        '#title_display' => 'invisible',
+        '#target_type' => $element['#target_type'],
+        '#selection_handler' => $element['#selection_handler'],
+        '#selection_settings' => $element['#selection_settings'],
+        '#validate_reference' => FALSE,
```

Comment:
```
    // This should only be set to FALSE if proper validation by the selection
    // handler is performed at another level on the extracted form values.
    $info['#validate_reference'] = TRUE;
```

Are we validating the references somewhere else?

Reply:
Yes. It should use with TRUE element. Originally it was inherited code of widget. But as we don't have Constraints, reference validation should be enabled.

Reply:
It's still there as far as I see...

Reply:
Fixed in other places as well.